src/features/activityFeed/state/activityFeedState.ts

```diff
@@ -18,16 +20,63 @@ export function selectLogCount (qriRef: QriRef): (state: RootState) => number {
   }
 }
 
+export function selectRunInfo (runID: string): (state: RootState) => RunLogInfoState {
+  return (state: RootState) => {
+    return state.activityFeed.runInfo[runID]
+  }
+}
+
+export interface RunLogInfoState {
+  error?: string
+  runLog?: Run
+}
+
 export interface ActivityFeedState {
   datasetLogs: Record<string, LogItem[]>
   loading: boolean
   error: ApiErr
+  runInfo: Record<string, RunLogInfoState>
 }
 
 const initialState: ActivityFeedState = {
   datasetLogs: {},
   loading: false,
-  error: NewApiErr()
+  error: NewApiErr(),
+  runInfo: {}
+}
+
+// reduces a RunLog to a newline-delimited string, used in run logs
+export function outputFromRunLog (runLog: Run): string {
+  let logOutput = ''
+  logOutput += `[run id     ] ${runLog.id}\n`
+  logOutput += `[started    ] ${runLog.startTime}\n`
+  logOutput += `[ended      ] ${runLog.stopTime}\n`
+  logOutput += `[status     ] ${runLog.status}\n`
+  logOutput += `[steps      ] ${runLog.steps.length} total\n`
+  logOutput += runLog.steps.reduce((acc, step, i) => {
+    let stepString = acc
+
+    stepString += `[step ${i}     ]\n`
+    stepString += `[name       ] '${step.name}'\n`
+    stepString += `[started    ] ${step.startTime}\n`
+    stepString += `[ended      ] ${step.stopTime}\n`
+    stepString += `[status     ] ${step.status}\n`
+
+    if (step.output) {
+      stepString += step.output.reduce((acc, output) => {
+        let outputString = acc
+        // TODO(chriswhong): calls to /auto/runinfo are returning Timestamp and Type keys, this needs to be fixed on the backend
```

Comment:
good find.

Asmir has a pr to fix this, but I wanted to give you a heads up, I'm advocating we use `timestamp` (rather than `Timestamp` &or `ts`) everywhere we use `events` in the backend codebase, which will require some front end refactoring when it goes live. Ultimately, it means we can use the same `EventLogLine` typescript interface everywhere we interact with events on the frontend, rather than only being able to use it when looking at events coming over the websocket connection.